src/OpenTelemetry/Trace/TracerProviderExtensions.cs

```diff
@@ -43,6 +43,52 @@ public static TracerProvider AddProcessor(this TracerProvider provider, BaseProc
             return provider;
         }
 
+        /// <summary>
+        /// Flushes the all the processors at TracerProviderSdk, blocks the current thread until flush
+        /// completed, shutdown signaled or timed out.
+        /// </summary>
+        /// <param name="provider">TracerProviderSdk instance on which ForceFlush will be called.</param>
+        /// <param name="timeoutMilliseconds">
+        /// The number of milliseconds to wait, or <c>Timeout.Infinite</c> to
+        /// wait indefinitely.
+        /// </param>
+        /// <returns>
+        /// Returns <c>true</c> when force flush succeeded; otherwise, <c>false</c>.
+        /// </returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">
+        /// Thrown when the <c>timeoutMilliseconds</c> is smaller than -1.
+        /// </exception>
+        /// <remarks>
+        /// This function guarantees thread-safety.
+        /// </remarks>
+        public static bool ForceFlush(this TracerProvider provider, int timeoutMilliseconds = Timeout.Infinite)
+        {
+            if (provider == null)
+            {
+                throw new ArgumentNullException(nameof(provider));
+            }
+
+            if (provider is TracerProviderSdk tracerProviderSdk)
```

Comment:
This seems like a hack. There should be some public type (class or interface) representing TracerProviderSdk. But I guess it's too late for that 😃

Reply:
Agree. :)
I just made it like it was done for `Shutdown` method. In my opinion, it's better to change nothing at the moment and make architectural changes at separate PR.

Reply:
Lets not address any other issue in this PR :)
Happy to address them separate. (or not address :D)